core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -189,23 +239,26 @@ private boolean insertAsSelect(
             MatViewRefreshState state,
             MatViewDefinition viewDef,
             TableWriterAPI tableWriter,
+            SampleByIntervalIterator intervalIterator,
             long baseTableTxn,
             long refreshTriggeredTimestamp
     ) throws SqlException {
         assert state.isLocked();
 
         final int maxRecompileAttempts = configuration.getMatViewMaxRecompileAttempts();
+        final int maxRetriesOnOom = configuration.getMatViewMaxRetriesOnOom();
         final long batchSize = configuration.getMatViewInsertAsSelectBatchSize();
 
         RecordCursorFactory factory = null;
         RecordToRowCopier copier;
+        int intervalStep = intervalIterator.getStep();
         long rowCount = 0;
         long refreshTimestamp = microsecondClock.getTicks();
         try {
             factory = state.acquireRecordFactory();
             copier = state.getRecordToRowCopier();
 
-            for (int i = 0; i < maxRecompileAttempts; i++) {
+            for (int i = 0, n = Math.max(maxRecompileAttempts, maxRetriesOnOom); i < n; i++) {
```

Comment:
do we really need two retry limits? why not simply use the common one?

Reply:
That's because the max number of recompilation attempts should be high enough, e.g. 10, while don't want to keep many attempts on OOM.

Reply:
cc @mtopolnik

Reply:
We should also consider what's easier for the user, proliferation of settings makes maintenance more complex. We can probably have a good single setting that will work for most cases, and then only add the distinction if we're forced to by a specific problem a user has.

Reply:
Then we should rename the property and update its documentation. It's no longer max number of recompile attempts.